src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -1611,7 +1574,7 @@ private ConsumeTokenResult ConsumeNextTokenMultiSegment(byte marker)
                 }
             }
 
-            if (!_isNotPrimitive)
+            if (CurrentDepth == 0)
```

Comment:
Combined with the changes to CurrentDepth, do you instead want _bitStack.CurrentDepth?

Reply:
I'll update that in the other PR once this is in: https://github.com/dotnet/corefx/pull/36116